tests/cnf/core/network/metallb/tests/bgp-gracefulrestart.go

```diff
@@ -205,6 +373,59 @@ func gracefulRestartTest(
 	Expect(err).ToNot(HaveOccurred(), "Unexpectedly succeeded in curling the Nginx pod")
 }
 
+func gracefulRestartTestWithBFD(
+	extTrafficPolicy corev1.ServiceExternalTrafficPolicyType,
+	destIPaddress,
+	ipv4metalLbIP string,
+	frrPod *pod.Builder,
+	ipAddressPool *metallb.IPAddressPoolBuilder,
+	workerLabelMap map[string]string) {
+	By("Creating a MetalLB service")
+	setupMetalLbService(netparam.IPV4Family, ipAddressPool, extTrafficPolicy)
+
+	By("Checking that routes have been added to the BGP table")
+	verifyRoutesAmount(frrPod, 1)
+
+	By("Running http check before the fail over test")
+	validateCurlToNginxPod(frrPod, ipv4metalLbIP, destIPaddress)
+
+	By("Deleting the MetalLB CR to remove all frr-k8s pods and simulate a GR scenario (e.g., during an upgrade)")
+	removeMetalLbCR()
+
+	By("Checking that BGP sessions are down")
+	verifyMetalLbBGPSessionsAreDown(frrPod, ipv4NodeAddrList)
+
+	By("Checking that routes exists in the bgp table")
```

Comment:
small note, if the above function call take more than 2 minutes, that might fail even if GR works as expected.
In the past we discussed that you might check the lifetime of a route?

Reply:
The above func  max can take 1 min (30 sec * 2).

Reply:
Currently this is the value true, but sometimes timeouts are changed as consider they are not picked on a specific value (maybe a comment)

Reply:
added